cmake/add_dcap_client_target.cmake

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+## This function adds a CMake target for the DCAP client and its dependencies provided through nuget
+## redistributable packages. This allows the caller to add a dependency on these redistributables so
+## that they will be copied to the output folder along with the target taking the dependency.
+## 
+## TARGET_NAME: Name of the target to add for the DCAP client redistributables that the caller will
+##              call add_dependency on. This should be unique for each caller.
+## 
+
+function(add_dcap_client_target TARGET_NAME)
+
+    if (NOT WIN32)
+        message(WARNING "import_dcap_client is only intended for WIN32 build environments. Check if this invocation is needed.")
+    endif ()
+
+    if (CMAKE_BUILD_TYPE MATCHES Debug OR CMAKE_BUILD_TYPE MATCHES RelWithDebugInfo)
+        set(IMPORT_BUILD_TYPE Debug)
+    elseif (CMAKE_BUILD_TYPE MATCHES Release OR CMAKE_BUILD_TYPE MATCHES MinSizeRel)
+        set(IMPORT_BUILD_TYPE Release)
+    else ()
+        message(FATAL_ERROR "Unsupported CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}, import_dcap_client needs to be updated to support it.")
+    endif ()
+
+    # Initialize the null list of dependencies for the target
+    set(DEPENDENCIES "")
+
+    # Define the DCAP provider paths
+    set(AZURE_DCAP_QUOTEPROV ${CMAKE_SOURCE_DIR}/prereqs/nuget/Microsoft.Azure.DCAP.Client/lib/${IMPORT_BUILD_TYPE}/dcap_quoteprov.dll)
+    set(AZURE_DCAP_QUOTEPROV_SYMBOLS ${CMAKE_SOURCE_DIR}/prereqs/nuget/Microsoft.Azure.DCAP.Client/lib/${IMPORT_BUILD_TYPE}/dcap_quoteprov.pdb)
+
+    # Note that for all of dcap_quoteprov's own dependencies, we always take the Release build.
+    # The Debug versions don't come with symbols anyway and take a dependency on the debug msvcr110d.dll
+    # which is not provided by the vcredist_x64 package and will cause runtime load faiures.
+    set(LIBCURL ${CMAKE_SOURCE_DIR}/prereqs/nuget/curl.redist/build/native/bin/v110/x64/Release/dynamic/libcurl.dll)
```

Comment:
I thought that all of these aren't needed anymore with the new DCAP provider?

Reply:
Version 0.0.1 does not require these anymore.

Reply:
Removed.